react-components/src/tools/EditTool.js

```diff
@@ -255,21 +387,36 @@ const DEFAULT_TOOL = {
     type: "",
 };
 
+const DEFAULT_ADMIN_TOOL = {
+    ...DEFAULT_TOOL,
+    implementation: {
+        implementor: "",
+        implementor_email: "",
+        test: {
+            input_files: [],
+            output_files: [],
+        },
+    },
+};
+
 function mapPropsToValues(props) {
-    const { tool } = props;
+    const { tool, isAdmin } = props;
     if (!tool) {
-        return { ...DEFAULT_TOOL };
+        return isAdmin ? { ...DEFAULT_ADMIN_TOOL } : { ...DEFAULT_TOOL };
     } else {
         return { ...tool };
     }
 }
 
 EditToolDialog.propTypes = {
     open: PropTypes.bool.isRequired,
+    isAdmin: PropTypes.bool.isRequired,
+    isAdminPublishing: PropTypes.bool.isRequired,
     presenter: PropTypes.shape({
-        addTool: PropTypes.func.isRequired,
-        updateTool: PropTypes.func.isRequired,
```

Comment:
We don't want these functions to be required?

Reply:
So for Manage Tools and Tool Admin, they require `addTool` and `updateTool`.  But for Tool Request, it only has `onPublish`. I realized you can add custom validation functions, so I'll add that for this.

Reply:
I forgot (or hadn't realized yet) that this form is also reused for Tool Requests. 👌